src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -380,6 +379,8 @@ BoundExpression visitArgumentsAndFinishRewrite(BoundCall node, BoundExpression?
                     rewrittenReceiver = null;
                 }
 
+                rewrittenReceiver = AdjustReceiverForExtensionsIfNeeded(rewrittenReceiver, method);
```

Comment:
>AdjustReceiverForExtensionsIfNeeded [](http://example.com/codeflow?start=36&length=35)

I think that whenever possible, this transformation should be done in the same code that uses ```ReferToTempIfReferenceTypeReceiver```. It looks like for this specific code path it will be possible. Basically, there should be one place where we transform receivers for a given language costruct.